commands/analyze.go

```diff
@@ -39,109 +39,142 @@ func init() {
 
 func analyze(inDb string, configFile string) error {
```

Comment:
The MetaDB class was refactored into the RITADatabaseIndex and RITADatabase classes. So most of the changes revolve around the refactor.

Reply:
What's the advantage of the refactor?

Reply:
The MetaDB was getting way too complicated, becoming a "god" class for keeping track of which databases have imported, analyzed, etc. as well as keeping track of which files were imported through the main importer. This PR adds significant functionality to the database index side of things. Following single responsibility principle, and the scout's rule, I figured I'd make this section of code a better place by splitting out the components.

Reply:
This seems like a huge number of changes for a single pull request.

Reply:
I agree. We should have separate PR's for the refactor and the analyze/import collision issue. We should create a separate issue for the refactor and only go with the changes needed for the analysis import issue for this pull request. When creating the issue for refactor make sure that we explain why each step is necessary and the goals of the overall issue. Since the MetaDB is so critical to RITA functioning and RITA is open source we should have good in line documentation for the community to reference while they might be working on RITA for their stuff

Reply:
@Zalgo2462 is there a clean way to break these apart? I know we'd miss a lot of the added error checking benefits, but can the needed changes for setting the import finished flag and checking it be added quickly on top of the existing master?  If so, we can merge that part in fairly quickly without waiting any longer to figure out styling and whatnot on the rest.  The refactor looks like it has plenty of merit, but it's just a lot more to go through and a lot more places to cause holdup in reviews if it's not necessary for the required functionality.  

And just to put on record since it was the first question I asked, the refactor didn't change the database schema in Mongo at all.  The only database change was adding a single boolean flag to support the analyze/import collision.

Reply:
@ethack I'll get a version in without the refactor / error checks in by the end of the day. 

The error checks could be added to the existing MetaDB class in a separate PR after that if we are interested.

Unfortunately, this refactor can't build on top of the additions to the MetaDB class as the whole MetaDB go file is removed in this PR.

I don't think there is an easy way to break this PR up as it splits the MetaDB class into three classes RITADBIndex, RITADatabase, and ImportedFilesIndex. 

The new RITADBIndex and RITADatabase objects contain new code to handle the import_finished flag. The ImportedFilesIndex is essentially what is left over after extracting RITADBIndex and RITADatabase out of the MetaDB class.